gateway/server.go

```diff
@@ -1428,19 +1438,25 @@ func setupPortsWhitelist() {
 	config.SetGlobal(globalConf)
 }
 
-func startServer() {
+type Gateway struct {
+	DefaultProxyMux *proxyMux
+	Port            int
```

Comment:
We also have a control port. Maybe just pass the whole configuration object instead?

Reply:
If we have a config field inside the gateway struct, we should also change all the gateway receiver from `func (gw Gateway)` to a pointer receiver`func (gw *Gateway)`, just in case we modify the config in any gateway func.